pytorch_lightning/loggers/mlflow.py

```diff
@@ -151,6 +152,9 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
             if isinstance(v, str):
                 log.warning(f'Discarding metric with string value {k}={v}.')
                 continue
+            # MLFlow does not support some special characters in metric name.
+            k = re.sub("[^a-zA-Z0-9_/. -]+", "", k)
```

Comment:
Maybe worth to put once a warning there, as we modify user logged keys.

Reply:
I ll update the request with a warnings.warn tomorrow. Don't know how to add a test but i ll have a look into it.

Reply:
Updated